### ISSUE TITLE
Deserialization of medium-list > track-count

### ISSUE THREAD

**Issue by wo80** [avatar29A/MusicBrainz#3]
I'm having an issue with the following code:
```c#
var query = "artist:(giant sand) release:(tucson)";
var result = Release.Search(Uri.EscapeDataString(query), 10);
```
It translates to the following url:
http://musicbrainz.org/ws/2/release?query=artist%3A%28giant%20sand%29%20release%3A%28tucson%29&limit=10&offset=0

The XML returned contains the following elements:
```xml
<medium-list count="1">
  <track-count>19</track-count>
  <medium>
    <format>CD</format>
    <disc-list count="1" />
    <track-list count="19" />
  </medium>
</medium-list>
```
But ```result[i].MediumList.TrackCount``` is always 0.

As far as I can see, the serialization attributes of ```Entities.Collections.MediumList``` are configured correctly. Any ideas?

**Comment by deepakyadav** [comment]
While deserializing a collection other properties are not deserialized.
I faced the same issue, then I used List<> as property instead of inheritance.

For code reference you can check my repo https://github.com/deepakyadav/MusicBrainz-Web-Service

-----Original Message-----

**Comment by wo80** [comment]
Works a treat. I think the changes should be backported...

**Comment by wo80** [comment]
Fixed in 9c0163739914fb9b7cc89b3613a735ccddd4fd56
